packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/create/iti38-envelope.ts

```diff
@@ -28,14 +29,18 @@ function createSoapBody(bodyData: OutboundDocumentQueryReq): object {
   const classCode = bodyData.classCode;
   const practiceSettingCode = bodyData.practiceSettingCode;
   const facilityTypeCode = bodyData.facilityTypeCode;
-  const serviceDateFrom = dayjs(bodyData.serviceDate?.dateFrom).format("YYYYMMDDHHmmss");
-  const serviceDateTo = dayjs(bodyData.serviceDate?.dateTo).format("YYYYMMDDHHmmss");
-  const documentCreationDateFrom = dayjs(bodyData.documentCreationDate?.dateFrom).format(
-    "YYYYMMDDHHmmss"
-  );
-  const documentCreationDateTo = dayjs(bodyData.documentCreationDate?.dateTo).format(
-    "YYYYMMDDHHmmss"
-  );
+  const serviceDateFrom = bodyData.serviceDate?.dateFrom
+    ? dayjs(bodyData.serviceDate.dateFrom).format(dateFormat)
+    : undefined;
+  const serviceDateTo = bodyData.serviceDate?.dateTo
+    ? dayjs(bodyData.serviceDate.dateTo).format(dateFormat)
+    : undefined;
+  const documentCreationDateFrom = bodyData.documentCreationDate?.dateFrom
+    ? dayjs(bodyData.documentCreationDate.dateFrom).format(dateFormat)
+    : undefined;
+  const documentCreationDateTo = bodyData.documentCreationDate?.dateTo
```

Comment:
Lots of repetitive code. Create a date-generating function and pass whatever should contain a date for all these instances